File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/string/ConcatenateTest.java
package com.fincity.nocode.kirun.engine.function.system.string;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonArray;
import com.google.gson.JsonPrimitive;

import reactor.test.StepVerifier;

class ConcatenateTest {

	@Test
	void test() {
		Concatenate catF = new Concatenate();
		var list = new JsonArray();
		list.add("no code ");
		list.add(" Kirun ");
		list.add(" true ");
		list.add("\"'this is between the strings qith special characters'\"");
		list.add(" PLATform ");

		StepVerifier
				.create(catF.execute(new ReactiveFunctionExecutionParameters(new KIRunReactiveFunctionRepository(),
						new KIRunReactiveSchemaRepository()).setArguments(Map.of("value", list))))
				.expectNextMatches(result -> result.next().getResult().get("value").equals(new JsonPrimitive(
						"no code  Kirun  true \"'this is between the strings qith special characters'\" PLATform ")))
				.verifyComplete();
	}

}

File: kirun-js/__tests__/engine/function/system/string/ConcatenateTest.ts
import { Concatenate } from '../../../../../src/engine/function/system/string/Concatenate';
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';
import { MapUtil } from '../../../../../src/engine/util/MapUtil';

const cat: Concatenate = new Concatenate();

test('conatenate test1', () => {
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters();

    let array: string[] = [];
    array.push('I ');
    array.push('am ');
    array.push('using ');
    array.push('eclipse ');
    array.push('to ');
    array.push('test ');
    array.push('the ');
    array.push('changes ');
    array.push('with ');
    array.push('test ');
    array.push('Driven ');
    array.push('developement');

    fep.setArguments(MapUtil.of('value', array));

    expect(cat.execute(fep).allResults()[0].getResult().get('value')).toBe(
        'I am using eclipse to test the changes with test Driven developement',
    );
});

test('conatenate test2', () => {
    let list: string[] = [];
    list.push('no code ');
    list.push(' Kirun ');
    list.push(' true ');
    list.push('"\'this is between the strings qith special characters\'"');
    list.push(' PLATform ');
    list.push('2');

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters();

    fep.setArguments(MapUtil.of('value', list));
    expect(cat.execute(fep).allResults()[0].getResult().get('value')).toBe(
        'no code  Kirun  true "\'this is between the strings qith special characters\'" PLATform 2',
    );
});
